Makefile

```diff
@@ -17,8 +17,12 @@ MAKE_PID := $(shell echo $$PPID)
 JOB_FLAG := $(filter -j%, $(subst -j ,-j,$(shell ps T | grep "^\s*$(MAKE_PID).*$(MAKE)")))
 JOBS := $(subst -j,,$(JOB_FLAG))
 
-release-targets: CPP_FLAGS := -O3 -Wall -c -fmessage-length=0 -fPIC -flto
-test-targets: CPP_FLAGS = -O0 -Wall -c -fmessage-length=0 -fPIC $(GCOV_FLAGS)
+# keep debug option only
+CPP_FLAGS := $(if $(findstring -g,$(CXXFLAGS)), -g)
```

Comment:
Why only the `-g` flag, and not all flags?

Reply:
Debian passes `CXXFLAGS` and `CFLAGS` as `-g -O2 -ffile-prefix-map=/sonic/src/linkmgrd=. -fstack-protector-strong -Wformat -Werror=format-security`, keeping the debug flag only here is to ensure that other flags hardcoded by `linkmgrd` will not be overridden, for instance, `linkmgrd` uses `-O3` for the release target.

Reply:
For `-O2` vs `-O3`, I believe the last specified flag takes effect, so if `-O3` is specified after `-O2`, then `-O3` gets used.

Reply:
Yes, keeping both optimization flag in the compile command is a little bit confusing somehow, as we only need `-g` in the flags, is it better if we only keep `-g` here?

Reply:
Technically, there are other security-related flags being passed down that are good to have (stack protector and format string checking).